### PR TITLE
Deprecation moved from WARN to STOP

### PR DESCRIPTION
**Pull request by millerg23**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by millerg23** [comment]
I have to still to remove `derive_vars_suppqual()` but have moved the others from WARNING to ERROR.
Can you check if these look correct?

File: R/derive_var_ady.R [hunk 1]
```diff
@@ -41,14 +41,5 @@
...
-  date <- assert_symbol(enquo(date))
-  assert_data_frame(dataset, vars(!!reference_date, !!date))
-  deprecate_warn("0.7.0", "derive_var_ady()", "derive_vars_dy()")
-
-  derive_vars_dy(
-    dataset,
-    reference_date = !!reference_date,
-    source_vars = vars(ADY = !!date)
-  )
+  deprecate_stop("0.8.0", "derive_var_ady()", "derive_vars_dy()")
```
Comment (thomas-neitmann) [inline R/derive_var_ady.R:44]: I actually think the first argument should stay at 0.7.0. According to the documentation the `when` argument is a "string giving the version when the behaviour was deprecated".

File: R/derive_var_atirel.R [hunk 2]
```diff
@@ -38,28 +38,5 @@
...
-  dataset %>%
-    mutate(!!new_var :=
-      case_when(
-        is.na(TRTSDTM) ~ NA_character_,
-        ASTDTM >= TRTSDTM ~ "CONCOMITANT",
-        !is.na(AENDTM) & AENDTM < TRTSDTM ~ "PRIOR",
-        date(ASTDTM) == date(TRTSDTM) & toupper(!!flag_var) %in% c("H", "M") ~ "CONCOMITANT",
-        TRUE ~ "PRIOR_CONCOMITANT"
-      ))
+  deprecate_stop("0.8.0", "derive_var_atirel()", "mutate()")
```
Comment (thomas-neitmann) [inline R/derive_var_atirel.R:41]: Please use the `details` argument to be a bit more specific.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

**Comment by millerg23** [comment]
Do I need to add a NEWS item for this?

File: R/derive_var_ady.R (hunk 1 above)
Comment (millerg23) [reply R/derive_var_ady.R:44]: OK I see, I thought it was the version when it was moved to a certain stage of the deprecation status.

**Comment by thomas-neitmann** [comment]
I'd say so. It should be explicit that using the functions now throws an error rather than just a warning. I think the proper term is that they are now "defunct" rather than just deprecated.

File: R/derive_var_ady.R (hunk 1 above)
Comment (millerg23) [reply R/derive_var_ady.R]: Fixed.

File: R/derive_var_atirel.R (hunk 2 above)
Comment (millerg23) [reply R/derive_var_atirel.R]: Have updated, let me know if OK?

File: R/derive_var_atirel.R [hunk 3]
```diff
@@ -38,28 +38,9 @@
...
-    sep = "\n"
-  ))
-
-  flag_var <- assert_symbol(enquo(flag_var))
-  assert_data_frame(dataset,
-    required_vars = vars(STUDYID, USUBJID, TRTSDTM, ASTDTM, AENDTM, !!flag_var)
+  deprecate_stop("0.7.0",
+                 "derive_var_atirel()",
+                 details = "Please use combination of `mutate()` and `case_when()` instead."
   )
```
Comment (thomas-neitmann) [inline R/derive_var_atirel.R:44]:
```suggestion
  deprecate_stop(
    "0.7.0",
    "derive_var_atirel()",
    details = "Please use combination of `mutate()` and `case_when()` instead."
  )
```

File: R/derive_var_extreme_flag.R [hunk 4]
```diff
@@ -199,6 +199,11 @@ derive_var_extreme_flag <- function(dataset,
                                     mode,
                                     filter = deprecated(),
                                     check_type = "warning") {
+
+  if (!missing(filter)) {
+    deprecate_stop("0.7.0", "derive_var_extreme_flag(filter = )", "restrict_derivation(filter = )")
+  }
```
Comment (thomas-neitmann) [inline R/derive_var_extreme_flag.R:205]:
```suggestion
  if (!missing(filter)) {
    deprecate_stop("0.7.0", "derive_var_extreme_flag(filter = )", "restrict_derivation(filter = )")
  }
```

File: R/derive_var_extreme_flag.R [hunk 5]
```diff
@@ -368,6 +351,10 @@ derive_var_worst_flag <- function(dataset,
                                   filter = deprecated(),
                                   check_type = "warning") {
 
+  if (!missing(filter)) {
+    deprecate_stop("0.7.0", "derive_var_worst_flag(filter = )", "restrict_derivation(filter = )")
+  }
+
```
Comment (thomas-neitmann) [inline R/derive_var_extreme_flag.R:357]:
```suggestion
  if (!missing(filter)) {
    deprecate_stop("0.7.0", "derive_var_worst_flag(filter = )", "restrict_derivation(filter = )")
  }

```

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

**Review by thomas-neitmann — APPROVED** [review]

**Comment by thomas-neitmann** [comment]
Thanks a lot @millerg23!

**Comment by millerg23** [comment]
News item added, and as discussed with @thomas-neitmann no action required for `derive_vars_suppqual()` as this went straight to `deprecation_stop` in 0.7 version, and should be removed in 0.9 version.

**Comment by bms63** [comment]
Can you fix code style action?

**Pull request merged**